docker/start.py

```diff
@@ -170,7 +170,7 @@ def run_generate_config(environ, ownership):
 
     if ownership is not None:
         # make sure that synapse has perms to write to the data dir.
-        subprocess.check_output(["chown", ownership, data_dir])
+        subprocess.check_output(["chown", ownership, data_dir, config_dir])
```

Comment:
it's not obvious to me that the synapse process should have permission to write to the config dir? they are supposed to be static files.

Could you expand a bit on your usecase and why you are hitting this?

Reply:
As mentioned in #7304 , this might be only an issue when generate is used for the config_dir at least. Otherwise the config already has write access from the process via data_dir. 

In fact. it might be worth actually make the separate config_dir default from a security point of view?